runners/spark/src/main/java/org/apache/beam/runners/spark/SparkPipelineResult.java

```diff
@@ -207,12 +202,19 @@ protected State awaitTermination(final Duration duration) throws ExecutionExcept
            terminationState = State.DONE;
            break;
          default:
-           state = State.UNKNOWN;
+           terminationState = State.UNKNOWN;
            break;
        }
        return terminationState;
     }
+  }
 
+  private void changeState(State newState) {
```

Comment:
Maybe `finalizeState` would be a better name ? it is used to "finalize" the state after some "done signal" - stop/cancel/timeout